tests/internal/aws_s3_local_buffer.c

```diff
@@ -0,0 +1,154 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+#include <fluent-bit/flb_sds.h>
+#include <fluent-bit/flb_s3_local_buffer.h>
+#include <fluent-bit/flb_output_plugin.h>
+#include <fluent-bit/flb_mem.h>
+#include <fluent-bit/flb_info.h>
+#include <fluent-bit/flb_aws_util.h>
+
+#include "flb_tests_internal.h"
+
+#define DIRECTORY "/fluent-bit/buffer/s3"
+#define PLUGIN_NAME "flb_s3_plugin"
+#define TEST_DATA "I love Fluent Bit"
+#define KEY_1 "key1"
+#define KEY_2 "key2"
+#define KEY_3 "key3"
+
+static void test_chunk_compare(struct flb_local_buffer *store, 
+                               struct flb_local_chunk *chunk,
+                               char *tag, char *data)
+{
+    char *buffered_data = NULL;
+    size_t buffer_size;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    int ret;
+    /* Ensure data retreived is same as that which was stored. */
+    mk_list_foreach_safe(head, tmp, &store->chunks) {
+        chunk = mk_list_entry(head, struct flb_local_chunk, _head);
+        if (chunk->key == tag){
+            ret = flb_read_file(chunk->file_path, &buffered_data, &buffer_size);
+            TEST_CHECK(ret == 0);
+            TEST_CHECK(strcmp(buffered_data, data) == 0);
+        } 
+    }
+}
+
+static void test_flb_buffer_put_create_chunk()
+{
+    int ret;
+    struct flb_local_buffer *store = flb_calloc(1, sizeof(struct flb_local_buffer));
+    struct flb_output_instance *out = flb_calloc(1, sizeof(struct flb_output_instance));
+
+    store->dir = DIRECTORY;
+    memcpy(out->name, PLUGIN_NAME, strlen(PLUGIN_NAME));
+    store->ins = out;
+    mk_list_init(&store->chunks);
```

Comment:
Just had one thought, after each function call in the library, check mk_list_size(&store->chunks) == 1 (or whatever the length should be).

Reply:
Done